### PR TITLE
upgrade to MSAL 4.15 and use new `GetAccounts(userflow)`

### PR DESCRIPTION
**Pull request by jennyf19**
@jmprieur thoughts on this api change to filter the cache by user flow?

### CODE REVIEW

**Review by jmprieur — APPROVED** [review]
LGTM. Thanks @jennyf19 
This is a much better developer experience !!

File: active-directory-b2c-wpf/TokenCacheHelper.cs [hunk 1]
```diff
@@ -1,33 +1,5 @@
-﻿//------------------------------------------------------------------------------
-//
-// Copyright (c) Microsoft Corporation.
-// All rights reserved.
```
Comment (henrik-me) [inline active-directory-b2c-wpf/TokenCacheHelper.cs:4]: not related to PR: why are copyright notices not used in this sample?

Comment (jennyf19) [reply active-directory-b2c-wpf/TokenCacheHelper.cs:4]: i meant to add the shortened version. thanks for noticing. :)

File: active-directory-b2c-wpf/MainWindow.xaml.cs [hunk 2]
```diff
@@ -101,10 +93,10 @@ private async void CallApiButton_Click(object sender, RoutedEventArgs e)
         {
             AuthenticationResult authResult = null;
             var app = App.PublicClientApp;
-            IEnumerable<IAccount> accounts = await App.PublicClientApp.GetAccountsAsync();
+            var accounts = await app.GetAccountsAsync(App.PolicySignUpSignIn);
             try
             {
-                authResult = await app.AcquireTokenSilent(App.ApiScopes, GetAccountByPolicy(accounts, App.PolicySignUpSignIn))
+                authResult = await app.AcquireTokenSilent(App.ApiScopes, accounts.FirstOrDefault())
```
Comment (jmprieur) [inline active-directory-b2c-wpf/MainWindow.xaml.cs:101]: This is the sample code as for Azure AD now!

File: active-directory-b2c-wpf/MainWindow.xaml.cs [hunk 3]
```diff
@@ -101,10 +93,10 @@ private async void CallApiButton_Click(object sender, RoutedEventArgs e)
         {
             AuthenticationResult authResult = null;
             var app = App.PublicClientApp;
-            IEnumerable<IAccount> accounts = await App.PublicClientApp.GetAccountsAsync();
+            var accounts = await app.GetAccountsAsync(App.PolicySignUpSignIn);
```
Comment (jmprieur) [inline active-directory-b2c-wpf/MainWindow.xaml.cs:98]: The only difference is the extra policy (in addition to the authority, of course)

**Review by jmprieur — APPROVED** [review]

**Comment by bgavrilMS** [comment]
Can we merge this into master instead of msalv3 ? Is there any point to msalv3?

**Comment by jmprieur** [comment]
No @bgavrilMS: master is msal 2.0.
I need to do some branch magic to move master->msal2 and msal-> to master and have master being the default branch
but I need to coordinate with the docs team first

**Comment by bgavrilMS** [comment]
Yes, I was thinking of deleting the current master and making msalv3 -> master

**Comment by jennyf19** [comment]
we need to sync with @mmacy first before merging

**Comment by mmacy** [comment]
Here are the docs we need to check:

```console
marsh@ubu:~/repos/azure-docs-pr/articles/active-directory-b2c$ grep -ril --include=*.md "active-directory-b2c-dotnet-desktop" | sort
code-samples.md
quickstart-native-app-desktop.md
tutorial-desktop-app.md
tutorial-desktop-app-webapi.md
```

Cc: @msmimart

**Comment by mmacy** [comment]
@jennyf19 @jmprieur @msmimart Doc updates are required only if the default branch is switched to something other than **msalv3**. I don't see any in-article updates related to the code changes in this PR, only the link to the archive .ZIP.

| Doc                                                                                                                        | Update required?                 | PR  |
|----------------------------------------------------------------------------------------------------------------------------|----------------------------------|-----|
| [code-samples](https://docs.microsoft.com/en-us/azure/active-directory-b2c/code-samples)                                   | No                               | N/A |
| [quickstart-native-app-desktop](https://docs.microsoft.com/en-us/azure/active-directory-b2c/quickstart-native-app-desktop) | Yes - links to **msalv3** branch |     |
| [tutorial-desktop-app](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-desktop-app)                   | Yes - links to **msalv3** branch |     |
| [tutorial-desktop-app-webapi](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-desktop-app-webapi)     | No                               |  N/A  |

**Comment by jennyf19** [comment]
sounds good. Thanks for checking @mmacy

**Comment by jennyf19** [comment]
@bgavrilMS FYI

**Pull request merged**